CI clang/CMake build fails: missing <stdlib.h> for abort() in lib/suid.c

**Title:** CI clang/CMake build fails: missing stdlib include for abort()

**Description:**

CI fails when building with **clang** using the **CMake packaging preset**.

**Context:**

* Job: linux_cmake / PRESET=packaging / VARIANT=client / COMPILER=clang

**Error:**
lib/suid.c:45:3: error: call to undeclared library function 'abort'
note: include the header <stdlib.h>

**Cause:**
`abort()` is used in lib/suid.c without including the required standard header.
Clang (C99+) does not allow implicit function declarations.

**Fix:**
Add `#include <stdlib.h>` to lib/suid.c.

**Notes:**
This is exposed by CMake-based CI builds with clang; GCC may not flag it.

--- a/lib/suid.c
+++ b/lib/suid.c
@@ -15,6 +15,7 @@ static char rcsid[] = "$Id$";
 
 #include <sys/types.h>
 #include <sys/stat.h>
+#include <stdlib.h>
 #include <unistd.h>
 
 #include "libxymon.h"
@@ -71,4 +72,3 @@ void drop_root_and_removesuid(char *fn)
 
 	drop_root();
 }
-

